flash/partition_info/partition_info.c

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2023 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <stdio.h>
+#include <pico/stdlib.h>
+#include <pico/bootrom.h>
+#include "boot/picobin.h"
+
+
+#define PART_LOC_FIRST(x) ( ((x) & PICOBIN_PARTITION_LOCATION_FIRST_SECTOR_BITS) >> PICOBIN_PARTITION_LOCATION_FIRST_SECTOR_LSB )
+#define PART_LOC_LAST(x)  ( ((x) & PICOBIN_PARTITION_LOCATION_LAST_SECTOR_BITS)  >> PICOBIN_PARTITION_LOCATION_LAST_SECTOR_LSB )
+
+#define PARTITION_LOCATION_AND_FLAGS_SIZE  2
+#define PARTITION_ID_SIZE                  2
+#define PARTITION_NAME_MAX                 127
```

Comment:
@will-v-pi Is it worth hoisting any of these defines into pico-sdk, or is it fine for each project to define them itself?

Reply:
It'll probably be sensible to add them to `picobin.h`, along with more partition table handling in `bootrom.c` - @kilograham might have opinions too?